beeai/agents/utils.py

```diff
@@ -87,15 +86,6 @@ async def run_tool(
     return result
 
 
-@asynccontextmanager
-async def redis_client(redis_url: str) -> AsyncGenerator[redis.Redis, None]:
-    client = redis.Redis.from_url(redis_url)
-    await client.ping()
-    try:
-        yield client
-    finally:
-        await client.aclose()
-
```

Comment:
I believe the `fix_await()` below is closely related to `redis_client()` and should be thus moved together with it (and used to wrap the redis methods, in the same way as in the agents).